util/microservice-sdk/js/lib/configManager/Manager.js

```diff
@@ -0,0 +1,37 @@
+const Parsers = require('./parsers');
+const { Reader, Writer } = require('./fileManager');
+const Merger = require('./Merger');
+
+/**
+ * Creates the configuration file ./config/<service>.conf. The precedence is (from higher to lower):
+ * - Environment variables
+ * - User config file
+ * - Default config file
+ *
+ * @param {string} service acronym for the service, should only contain letters/numbers
+ * @param {string} path path to the user configuration file, defaults to './config'
+ */
+const createConfig = (service, path = './config', userConfigFile = 'production.conf') => {
+  const envVarsData = Parsers.EnvVars.parseEnvironmentVariables(service);
+  const userData = Reader.readUserConfig(path, userConfigFile);
+  const defaultData = Reader.readDefaultConfig();
+
+  const config = Merger.mergeConfigs(envVarsData, userData, defaultData);
+
+  Writer.writeJson(service, path, config);
+};
+
+/**
+ * Retrieves the configuration from the file.
+ *
+ * @param {string} service acronym for the service, should only contain letters/numbers
+ * @param {string} path path to the user configuration file, defaults to './config'
+ *
+ * @returns {{}} the configuration object
+ */
+const getConfig = (service, path = './config') => {
+  const config = Reader.readJson(service, path);
+  return config;
+};
```

Comment:
Again, I find it more practical to keep these settings in a cache in memory, so you avoid the I/O time to read the _.json_ file and we can also use `ConfigManager` directly as a cache to retrieve the settings at any point in our code.

Reply:
I don't think the IO time really matters here. The file is not a large one, and it is not requested a lot in the program. The configuration is usually read in the beginning and then it is forgotten or some of the parameters are stored in some class(es).

I did it this way to prevent the creation of a singleton, since it is the easiest way to do what you said, but the worst when it comes to testing and debugging. An improvement I can see now is to reutilize the JSON file if it is present, so we can create a volume in K8S for the `./config` directory and reutilize it between pod deaths. What do you think?

Reply:
I don't think a volume is necessary. Either way, the file will be regenerated whenever the application goes up (when it passes through the `ConfigManager.createConfig()` function). Leave it as it is, it is already very good.

Reply:
Yes, but a improvement would be to pass a variable to enable JSON reutilization. But those are little details that we can change in the future without problems.